okhttp/src/main/kotlin/okhttp3/internal/socket/OkioSocket.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package okhttp3.internal.socket
+
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.net.Socket
+import java.net.SocketAddress
+import okhttp3.internal.peerName
+import okio.BufferedSink
+import okio.BufferedSource
+import okio.Closeable
+import okio.buffer
+import okio.sink
+import okio.source
+
+interface OkioSocket : Closeable {
+  val source: BufferedSource
+  val sink: BufferedSink
+
+  var soTimeout: Int
```

Comment:
Should we avoid this API here? Does it map to Okio timeout? or should we leave it for code that knows it's a platform socket?

Reply:
Hmm. In theory any implementation of `OkioSocket` could enforce this timeout, even without a platform socket under the hood, by modifying whatever is being buffered by `source`. But I'm not sure if that makes for a good API or not 🤔